libtenzir/builtins/formats/syslog.cpp

```diff
@@ -10,41 +10,156 @@
 #include <tenzir/format/reader_factory.hpp>
 #include <tenzir/format/syslog.hpp>
 #include <tenzir/plugin.hpp>
+#include <tenzir/series_builder.hpp>
 #include <tenzir/table_slice_builder.hpp>
 #include <tenzir/to_lines.hpp>
 
 namespace tenzir::plugins::syslog {
 
 namespace {
 
+struct syslog_builder {
+public:
+  using message_type = format::syslog::message;
+
+  syslog_builder(const type& ty) : builder_(ty) {
+  }
+
+  auto add(const message_type& msg) -> bool {
+    record r{
+      {"facility", msg.hdr.facility},     {"severity", msg.hdr.severity},
+      {"version", msg.hdr.version},       {"timestamp", msg.hdr.ts},
+      {"hostname", msg.hdr.hostname},     {"app_name", msg.hdr.app_name},
+      {"process_id", msg.hdr.process_id}, {"message_id", msg.hdr.msg_id},
+      {"structured_data", msg.data},      {"message", msg.msg},
+    };
+    return builder_.try_data(r).operator bool();
+  }
+
+  auto rows() -> size_t {
+    return static_cast<size_t>(builder_.length());
+  }
+
+  auto finish() -> std::vector<table_slice> {
+    return builder_.finish_as_table_slice();
+  }
+
+private:
+  series_builder builder_;
+};
+
+struct legacy_syslog_builder {
+public:
+  using message_type = format::syslog::legacy_message;
+
+  legacy_syslog_builder(const type& ty) : builder_(ty) {
+  }
+
+  auto add(const message_type& msg) -> bool {
+    return builder_.add(msg.facility, msg.severity, msg.timestamp, msg.host,
+                        msg.app_name, msg.process_id, msg.content);
+  }
+
+  auto rows() -> size_t {
+    return builder_.rows();
+  }
+
+  auto finish() -> std::vector<table_slice> {
+    return {builder_.finish()};
+  }
```

Comment:
This and the one below needs to check whether there are any rows at all, as we'll otherwise yield empty table slices in the logic below.